### PR TITLE
[16.0][MIG] sale_automatic_workflow

### PR DESCRIPTION
**Pull request by atchuthan**
Removed dependency of sale_delivery_state(https://github.com/OCA/sale-workflow/pull/2280)

cc: @SodexisTeam

### CODE REVIEW

File: sale_automatic_workflow/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "name": "Sale Automatic Workflow",
+    "version": "16.0.1.0.1",
+    "category": "Sales Management",
+    "license": "AGPL-3",
+    "author": "Akretion, "
+    "Camptocamp, "
+    "Sodexis, "
+    "Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/sale-workflow",
+    "depends": ["sale_stock", "sales_team", "sale_delivery_state"],
```
Comment (RodrigoBM) [inline sale_automatic_workflow/__manifest__.py]: The sale_delivery_state dependency is no longer required.
Now in v16 we have the [delivery_status](https://github.com/OCA/OCB/blob/16.0/addons/sale_stock/models/sale_order.py#L34) field of the sale_stock module.

File: sale_automatic_workflow/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        string="Automatic Workflow",
+        ondelete="restrict",
+    )
+    all_qty_delivered = fields.Boolean(
+        compute="_compute_all_qty_delivered",
+        string="All quantities delivered",
+        store=True,
+    )
+
+    @api.depends("delivery_state")
```
Comment (RodrigoBM) [inline sale_automatic_workflow/models/sale_order.py]: Change delivery_state to delivery_status.
And we will set all_qty_delivered to true when delivery_status is full
```
@api.depends("delivery_status")
def _compute_all_qty_delivered(self):
    for order in self:
        order.all_qty_delivered = order.delivery_status == "full"
```

**Review by RodrigoBM — CHANGES_REQUESTED** [review]

**Comment by atchuthan** [comment]
@RodrigoBM Removed dependency of sale_delivery_state and changes made for above corrections.  Please review once again.  Thanks

**Review by rousseldenis — APPROVED** [review]
Code review

File: sale_automatic_workflow/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            sales_keep_order_date = self.filtered(
+                lambda sale: sale.workflow_process_id.invoice_date_is_order_date
+            )
+            if sales_keep_order_date:
+                new_vals = vals.copy()
+                del new_vals["date_order"]
+                res = super(SaleOrder, sales_keep_order_date).write(new_vals)
+                res |= super(SaleOrder, self - sales_keep_order_date).write(vals)
+                return res
+        return super(SaleOrder, self).write(vals)
```
Comment (RodrigoBM) [inline sale_automatic_workflow/models/sale_order.py]: Apply PEP 3135 (28-Apr-2007) to "super()."

File: sale_automatic_workflow/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,23 @@
+Create workflows with more or less automatization and apply it on sales
+orders.
+
+A workflow can:
+
+- Apply default values:
+
+  * Packing Policy (partial, complete)
```
Comment (RodrigoBM) [inline sale_automatic_workflow/readme/DESCRIPTION.rst]: Delete this point, it is not covered by the code.

File: sale_automatic_workflow/readme/DESCRIPTION.rst [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  * Shipping Policy (prepaid, manual, postpaid, picking)
+  * Invoice On (ordered quantities, shipped quantities)
```
Comment (RodrigoBM) [inline sale_automatic_workflow/readme/DESCRIPTION.rst]: Delete this point, it is not covered by the code.

File: sale_automatic_workflow/readme/DESCRIPTION.rst [hunk 6]
```diff
@@ -0,0 +1,23 @@
+Create workflows with more or less automatization and apply it on sales
+orders.
+
+A workflow can:
+
+- Apply default values:
+
+  * Packing Policy (partial, complete)
+  * Shipping Policy (prepaid, manual, postpaid, picking)
```
Comment (RodrigoBM) [inline sale_automatic_workflow/readme/DESCRIPTION.rst]: Change this to: Shipping Policy (Deliver each product when available or Deliver all products at once)

**Review by RodrigoBM — CHANGES_REQUESTED** [review]

File: sale_automatic_workflow/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2011 Akretion Sébastien BEAU <sebastien.beau@akretion.com>
+# Copyright 2013 Camptocamp SA (author: Guewen Baconnier)
+# Copyright 2016 Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Sale Automatic Workflow",
+    "version": "16.0.1.0.1",
```
Comment (RodrigoBM) [inline sale_automatic_workflow/__manifest__.py]: the version should be 16.0.1.0.0

**Review by RodrigoBM — CHANGES_REQUESTED** [review]

**Comment by atchuthan** [comment]
@RodrigoBM above requested changes are done.  Please review

**Review by amkarthik — APPROVED** [review]

**Review by RodrigoBM — APPROVED** [review]
LGFM! thanks for your work @atchuthan

**Comment by RodrigoBM** [comment]
@rousseldenis can you merge this?

thanks

**Pull request merged**